src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Parsing/ConnectionString/OdbcConnectionStringParser.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            var host = ParseHost();
+            if (host != null) host = ConnectionStringParserHelper.NormalizeHostname(host, utilizationHostName);
+            var portPathOrId = ParsePortPathOrId();
+            var databaseName = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _databaseNameKeys)?.Value;
+            var instanceName = ParseInstanceName();
+            return new ConnectionInfo(host, portPathOrId, databaseName, instanceName);
+        }
+
+        private string ParseHost()
+        {
+            var host = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _hostKeys)?.Value;
+            if (host == null) return null;
+
+            // Example of want we would need to process: win-database.pdx.vm.datanerd.us,1433\SQLEXPRESS
+            try
+            {
+                var splitIndex = host.IndexOf(StringSeparators.CommaChar);
+                if (splitIndex == -1) splitIndex = host.IndexOf(StringSeparators.BackslashChar);
+                host = splitIndex == -1 ? host : host.Substring(0, splitIndex);
+            }
+            catch
+            {
+                return null;
+            }
+            var endOfHostname = host.IndexOf(StringSeparators.ColonChar);
+            return endOfHostname == -1 ? host : host.Substring(0, endOfHostname);
+        }
+
+        private string ParsePortPathOrId()
+        {
+            // Some ODBC drivers use the "port" key to specify the port number
+            var portPathOrId = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _portKeys)?.Value;
+            if (!string.IsNullOrWhiteSpace(portPathOrId))
+            {
+                return portPathOrId;
+
+            }
+
+            // Some ODBC drivers include the port in the "server" or "data source" key
+            portPathOrId = ConnectionStringParserHelper.GetKeyValuePair(_connectionStringBuilder, _hostKeys)?.Value;
+            if (portPathOrId == null) return null;
+
+            try
+            {
+                if (portPathOrId.IndexOf(StringSeparators.ColonChar) != -1)
+                {
+                    var startOfValue = portPathOrId.IndexOf(StringSeparators.ColonChar) + 1;
+                    var endOfValue = portPathOrId.Length;
+                    return (startOfValue > 0) ? portPathOrId.Substring(startOfValue, endOfValue - startOfValue) : null;
+                }
+                if (portPathOrId.IndexOf(StringSeparators.CommaChar) != -1)
+                {
+                    var startOfValue = portPathOrId.IndexOf(StringSeparators.CommaChar) + 1;
+                    var endOfValue = portPathOrId.Contains(StringSeparators.BackslashChar)
+                        ? portPathOrId.IndexOf(StringSeparators.BackslashChar)
+                        : portPathOrId.Length;
+                    return (startOfValue > 0) ? portPathOrId.Substring(startOfValue, endOfValue - startOfValue) : null;
```

Comment:
The ternary here (and on line 74 above) isn't necessary -- `startOfValue` will always be > 0.